### PR TITLE
[expo-updates] useUpdateEvents hook

### PR DESCRIPTION
**Pull request by douglowder**
# Why

A React hook to make it more convenient to use `Updates.addListener` to listen for events coming from native code that checks for updates on launch.

# How

New file `UpdatesHooks.ts` containing the new hook. API doc comments include an example.

# Test Plan

- Test the new API in `App-apitest.js`. This example app is enhanced significantly to exercise the Updates API interactively. Eventually I will use it as the basis for additions to the updates E2E tests that will test the API automatically with Detox.
- Check generated unversioned API doc addition

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by wschurman — COMMENTED** [review]
LGTM but I'd wait for review from someone more familiar with the SDK design patterns.

**Comment by douglowder** [comment]
Thanks @wschurman -- it looks similar in pattern to the existing `addVisibilityListener` and `useVisibility` in `expo-navigation-bar`. See [this doc](https://docs.expo.dev/versions/unversioned/sdk/navigation-bar/#usevisibility). I think the one thing that might be good is to add an example similar to what I see in that module.

File: packages/expo-updates/CHANGELOG.md
Comment (tsapeta) [inline packages/expo-updates/CHANGELOG.md:9]:
```suggestion
- New `useUpdateEvents` hook. ([#21258](https://github.com/expo/expo/pull/21258) by [@douglowder](https://github.com/douglowder))
```

**Review by tsapeta — APPROVED** [review]

**Pull request merged**
